### PR TITLE
Updated link on Release Managers page

### PR DESCRIPTION
**Pull request by kadtendulkar**
Follow up on #35604

Updated the page [Release Managers](https://kubernetes.io/releases/release-managers/)

### CODE REVIEW

**Review by Sea-n — COMMENTED** [review]
Confirmed.
/lgtm

**Comment by kadtendulkar** [comment]
Hello @Sea-n , please approve PR.

**Comment by Sea-n** [comment]
Hi @kadtendulkar,

I am not a eligible approver for this PR (need to be `sig-docs-en-owners` usually).

My approval doesn't counts for @k8s-ci-robot, but I would be happy to try.
/approve

**Comment by reylejano** [comment]
This PR looks good but would like SIG Release to review
/sig release
/approve
/hold

@kubernetes/sig-release , ptal and remove the hold if this PR looks good

File: content/en/releases/release-managers.md
Comment (puerco) [inline content/en/releases/release-managers.md:214]:
```suggestion
[handbook-packaging]: https://git.k8s.io/release/hack/rapture/README.md
```

**Review by puerco — CHANGES_REQUESTED** [review]
Thanks for fixing this @kadtendulkar !
Just a tiny nit, let's keep the link in line with the others using the k8s.io redirector service:

File: content/en/releases/release-managers.md
Comment (reylejano) [reply content/en/releases/release-managers.md:214]: +1 to @puerco 's suggestion

**Comment by puerco** [comment]
Thanks @kadtendulkar !
/lgtm
/approve
/hold cancel

**Pull request merged**
